### PR TITLE
[libcxx] renames local variable to avoid shadowing werror

### PR DESCRIPTION
**Pull request by cjdb**
Due to the inclusion of a header, a global type is was being shadowed, which upset GCC.

### CODE REVIEW

**Review by philnik777 — APPROVED** [review]
LGTM assuming GCC is now happy.

File: libcxx/test/std/ranges/range.adaptors/range.join/range.join.sentinel/ctor.other.pass.cpp [hunk 1]
```diff
@@ -71,10 +71,10 @@ constexpr bool test() {
         BufferView<forward_iterator<const Inner*>, sentinel_wrapper<forward_iterator<const Inner*>>,
                    bidirectional_iterator<Inner*>, sentinel_wrapper<bidirectional_iterator<Inner*>>>;
     using JoinView = std::ranges::join_view<ConstInconvertibleOuter>;
-    using sentinel = std::ranges::sentinel_t<JoinView>;
+    using sentinel_t     = std::ranges::sentinel_t<JoinView>;
     using const_sentinel = std::ranges::sentinel_t<const JoinView>;
```
Comment (AdvenamTacet) [inline libcxx/test/std/ranges/range.adaptors/range.join/range.join.sentinel/ctor.other.pass.cpp:75]:
```suggestion
    using const_sentinel_t = std::ranges::sentinel_t<const JoinView>;
```

For consistency.

File: libcxx/test/std/ranges/range.adaptors/range.join/range.join.sentinel/ctor.other.pass.cpp [hunk 2]
```diff
@@ -71,10 +71,10 @@ constexpr bool test() {
...
-    static_assert(!std::constructible_from<sentinel, const_sentinel>);
-    static_assert(!std::constructible_from<const_sentinel, sentinel>);
+    static_assert(!std::constructible_from<sentinel_t, const_sentinel>);
+    static_assert(!std::constructible_from<const_sentinel, sentinel_t>);
```
Comment (AdvenamTacet) [inline libcxx/test/std/ranges/range.adaptors/range.join/range.join.sentinel/ctor.other.pass.cpp:77]:
```suggestion
    static_assert(!std::constructible_from<sentinel_t, const_sentinel_t>);
    static_assert(!std::constructible_from<const_sentinel_t, sentinel_t>);
```

**Review by ldionne — APPROVED** [review]
LGTM, thanks for fixing. I'll merge this now to make the CI green again ASAP.

**Pull request merged**